lib/html-min.cls.php

```diff
@@ -166,9 +170,27 @@ public function process()
 		return $this->_html;
 	}
 
+	/**
+	 * From LSCWP 6.2: Changed the function to test for special comments that will be skipped. See: https://github.com/litespeedtech/lscache_wp/pull/622
+	 */
 	protected function _commentCB($m)
 	{
-		return (0 === strpos($m[1], '[') || false !== strpos($m[1], '<!['))
+		// Check if is IE conditional comment
+		$is_ie_comment = (0 === strpos($m[1], '[') || false !== strpos($m[1], '<!['));
+
+		// Check if comment text is present in Page Optimization -> HTML Settings -> HTML Keep comments
+		$is_skip_comment = false;
+		if(count($this->_skipComments) > 0){
+			foreach ($this->_skipComments as $comment) {
+				if (!empty($comment) && strpos($m[1], $comment) !== false && $is_skip_comment === false) {
+					$is_skip_comment = true;
+					break;
+				}
+			}
+		}
+
+		// If comment is IE Conditional OR Comment that need to be kept, it will return the comment. Else it will be removed.
+		return ($is_ie_comment || $is_skip_comment)
```

Comment:
For $is_ie_comment one, it doesn't have to return here. It can be returned at line 179 directly.
Always keep code simple for easier reading.

Reply:
moved